### PR TITLE
feat: [#441] Optimize the HTTP testing process [#2] 

### PR DESCRIPTION
**Pull request by kkumar-gcc**
## 📑 Description

RelatedTo https://github.com/goravel/goravel/issues/441

## Summary by CodeRabbit

- **New Features**
	- Introduced a new interface `AssertableJSON` for flexible JSON representation.
	- Enhanced `TestRequest` interface with methods for various HTTP request types and context management.
	- Expanded `TestResponse` interface with new methods for response verification and content handling.
	- Added `AssertableJSONString` type for handling JSON strings and decoding.

- **Bug Fixes**
	- Improved error handling for JSON parsing in `AssertableJSONString`.

- **Documentation**
	- Updated method signatures for clarity and consistency in `TestResponse`.

## ✅ Checks

- [ ] Added test cases for my code

### CODE REVIEW

**Comment by kkumar-gcc** [comment]
@coderabbitai review

File: contracts/testing/test_request.go [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
+
 type TestRequest interface {
 	WithHeaders(headers map[string]string) TestRequest
 	WithHeader(key, value string) TestRequest
 	WithoutHeader(key string) TestRequest
 	WithCookies(cookies map[string]string) TestRequest
 	WithCookie(key, value string) TestRequest
+	WithContext(ctx context.Context) TestRequest
 	Get(uri string) (TestResponse, error)
+	Post(uri string, body io.Reader) (TestResponse, error)
```
Comment (hwbrzzl) [inline contracts/testing/test_request.go:16]: `body io.Reader` is a bit difficult to build. Can we provide a simple way to do so?

image

**Review by hwbrzzl — COMMENTED** [review]
Great PR 👍  just one question.

File: contracts/testing/test_request.go (hunk 1 above)
Comment (kkumar-gcc) [reply contracts/testing/test_request.go:16]: Sure, we can provide a interface to write body.

Comment (hwbrzzl) [reply contracts/testing/test_request.go:16]: We can optimize it in another PR if you want.

**Review by hwbrzzl — APPROVED** [review]

**Pull request merged**